tests/infra/network.py

```diff
@@ -483,14 +483,62 @@ def recover(self, args):
     def ignore_errors_on_shutdown(self):
         self.ignoring_shutdown_errors = True
 
+    def stop_node(self, node, primary=None):
+        if primary is None:
+            primary, _ = self.find_primary()
```

Comment:
This doesn't seem ideal, because we may not always know who's primary on shutdown, and the error cases where we don't may be interesting anyway. This is also done while other nodes are still running, and is potentially incomplete as a result.

I think we'd be better off doing this once all the nodes are stopped, by taking the longest ledger and checking everything else is a subset. Errors will only show up if there's a problem in the logic, or a fork.

Reply:
As discussed, this is now addressed by selecting the node with the longest ledger and comparing other nodes against that instead.